### DISCUSSION TITLE
DB engine Error while performing custom viz plugin in apache superset

### DISCUSSION THREAD

**Discussion by pm20202** [apache/superset#18944] [Q&A / Help]
I am trying to install Superset for building custom visualization using the guide at:
https://superset.apache.org/docs/contributing/creating-viz-plugins
Please note that I am attempting it on Ubuntu 20.04 VMs behind a proxy. And I am running superset on http://localhost:9000/(in dev mode)
After following all the steps successfully when I run it on 9000 port i am able to see Hello World chart but when I am trying to use this chart on dataset it shows me this error:

**DB engine Error
No module named 'superset.common.query_context_factory'
This may be triggered by:
Issue 1011 - Superset encountered an unexpected error.**

### Expected results

Hello World chart working on datasets.

### Actual results

After able to see Hello World plugin chart when trying to use this chart on dataset showing this error.
DB engine Error
No module named 'superset.common.query_context_factory'
This may be triggered by:
Issue 1011 - Superset encountered an unexpected error.

### Environment

(please complete the following information):

OS: Ubuntu Server 20.04
Python on VM: 3.8.10

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

I have checked issue code 1011 given in apache superset documentation but not able to find my mistake where i am going wrong.

**Comment by srinify** [comment]
Hi @pm20202 we're reserving Github issues primarily for bugs and occasional enhancement requests. I'm converting this to a Githhub Discussion, which is better suited for debugging. I'd also encourage you to join the Superset Slack community!

superset.apache.org/community
